accounts/checkers.go

```diff
@@ -0,0 +1,407 @@
+	)
+
+	GetNodeInfoPassThrough = rpcmiddleware.NewPassThrough(
+		&lnrpc.NodeInfoRequest{}, &lnrpc.NodeInfo{},
+	)
+
+	PendingChannelsEmptyRewriter = rpcmiddleware.NewResponseRewriter(
+		&lnrpc.PendingChannelsRequest{},
+		&lnrpc.PendingChannelsResponse{},
+		func(r *lnrpc.PendingChannelsResponse) (proto.Message, error) {
+			r.PendingOpenChannels = []*lnrpc.PendingChannelsResponse_PendingOpenChannel{}
+			r.PendingForceClosingChannels = []*lnrpc.PendingChannelsResponse_ForceClosedChannel{}
+			r.WaitingCloseChannels = []*lnrpc.PendingChannelsResponse_WaitingCloseChannel{}
+			r.TotalLimboBalance = 0
+
+			return r, nil
+		},
+	)
+
+	ListChannelsEmptyRewriter = rpcmiddleware.NewResponseRewriter(
+		&lnrpc.ListChannelsRequest{}, &lnrpc.ListChannelsResponse{},
+		func(r *lnrpc.ListChannelsResponse) (proto.Message, error) {
+			r.Channels = []*lnrpc.Channel{}
+
+			return r, nil
+		},
+	)
+
+	ClosedChannelsEmptyRewriter = rpcmiddleware.NewResponseRewriter(
+		&lnrpc.ClosedChannelsRequest{}, &lnrpc.ClosedChannelsResponse{},
+		func(r *lnrpc.ClosedChannelsResponse) (proto.Message, error) {
+			r.Channels = []*lnrpc.ChannelCloseSummary{}
+
+			return r, nil
+		},
+	)
+
+	WalletBalanceEmptyRewriter = rpcmiddleware.NewResponseRewriter(
+		&lnrpc.WalletBalanceRequest{},
+		&lnrpc.WalletBalanceResponse{},
+		func(t *lnrpc.WalletBalanceResponse) (proto.Message, error) {
+			t.ConfirmedBalance = 0
+			t.TotalBalance = 0
+			t.UnconfirmedBalance = 0
+			for acctName := range t.AccountBalance {
+				t.AccountBalance[acctName].ConfirmedBalance = 0
+				t.AccountBalance[acctName].UnconfirmedBalance = 0
+			}
+
+			return t, nil
+		},
+	)
+
+	GetTransactionsEmptyRewriter = rpcmiddleware.NewResponseRewriter(
+		&lnrpc.GetTransactionsRequest{},
+		&lnrpc.TransactionDetails{},
+		func(r *lnrpc.TransactionDetails) (proto.Message, error) {
+			r.Transactions = make([]*lnrpc.Transaction, 0)
+
+			return r, nil
+		},
+	)
+
+	ListPeersEmptyRewriter = rpcmiddleware.NewResponseRewriter(
+		&lnrpc.ListPeersRequest{}, &lnrpc.ListPeersResponse{},
+		func(r *lnrpc.ListPeersResponse) (proto.Message, error) {
+			r.Peers = []*lnrpc.Peer{}
+
+			return r, nil
+		},
+	)
+)
+
+func (s *Service) generateCheckers(
+	acct *OffChainBalanceAccount) map[string]rpcmiddleware.RoundTripChecker {
+
+	return map[string]rpcmiddleware.RoundTripChecker{
+		// Invoices:
+		"/lnrpc.Lightning/AddInvoice": rpcmiddleware.NewResponseRewriter(
+			&lnrpc.Invoice{},
+			&lnrpc.AddInvoiceResponse{},
+			func(t *lnrpc.AddInvoiceResponse) (proto.Message,
+				error) {
+
+				hash, err := lntypes.MakeHash(t.RHash)
+				if err != nil {
+					return nil, fmt.Errorf("error parsing "+
+						"invoice hash: %v", err)
+				}
+
+				return nil, s.Store.associateInvoice(
+					acct.ID, hash,
+				)
+			},
+		),
+		"/lnrpc.Lightning/ListInvoices": rpcmiddleware.NewResponseRewriter(
+			&lnrpc.ListInvoiceRequest{},
+			&lnrpc.ListInvoiceResponse{},
+			func(t *lnrpc.ListInvoiceResponse) (proto.Message,
```

Comment:
I think that's okay or even necessary to support pagination. Otherwise apps that paginate with the help of these indexes will do infinite requests? With the indexes intact, it just looks to the app like we deleted a lot of invoices because there are many gaps. But with the correct logic the app should still be able to paginate correctly.

Reply:
ah yes, good point 👍